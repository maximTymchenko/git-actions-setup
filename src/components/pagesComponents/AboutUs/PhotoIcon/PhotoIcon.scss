@import '../../../../styles/vars';

.photo-icon {
    margin-right: 25px;

    &:last-child {
        margin-right: 0;
    }
    &__wrapper {
        width: 270px;
        height: 270px;
        border: 2px solid $secondary;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    &__container {
        display: flex;
        overflow: hidden;
        margin: 50px 0 30px;
        overflow-x: scroll;
        width: 100%;
        max-width: 1000px;

        &::-webkit-scrollbar {
            height: 5px;
        }

        &::-webkit-scrollbar-track {
            box-shadow: inset 0 0 5px rgba(92, 92, 92, 0.2);
            border-radius: 20px;
        }

        &::-webkit-scrollbar-thumb {
            background-color: $primary__light;
            border-radius: 20px;
        }
    }

    &__img {
        // display: block;
        width: 220px;
        height: 220px;
    }

    &__name {
        font-family: 'OktaNeue-Black';
        font-size: 28px;
        font-weight: 900;
        color: $text;
        display: block;
        margin-left: 10px;
        margin-top: 15px;
    }

    &__position {
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        color: $text__body;
        margin-left: 10px;
    }
}
