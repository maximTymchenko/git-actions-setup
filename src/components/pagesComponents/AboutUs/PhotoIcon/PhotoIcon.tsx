import React, { FC } from 'react';
import './PhotoIcon.scss';

const PhotoIcon: FC = (props) => {
    return (
        <div data-testid="photo-icon" className="photo-icon__container">
            {props.ourTeam.map((team) => (
                <div key={team.id} className="photo-icon">
                    <div className="photo-icon__wrapper">
                        <img className="photo-icon__img" src={team.img} alt="CEO" />
                    </div>
                    <span className="photo-icon__name">{team.name}</span>
                    <span className="photo-icon__position">{team.position}</span>
                </div>
            ))}
        </div>
    );
};

export default PhotoIcon;
