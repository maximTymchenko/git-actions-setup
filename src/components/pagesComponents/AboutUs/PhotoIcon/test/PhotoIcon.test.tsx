import React from 'react';
import PhotoIcon from '../PhotoIcon';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

const ourTeam: TOurTeam = [
    { id: 0, name: 'Yurii Drozd', position: 'CEO & Managing Director', img: CEO },
    { id: 1, name: 'Yurii Drozd', position: 'CEO & Managing Director', img: CEO },
    { id: 2, name: 'Yurii Drozd', position: 'CEO & Managing Director', img: CEO },
    { id: 3, name: 'Yurii Drozd', position: 'CEO & Managing Director', img: CEO },
    { id: 4, name: 'Yurii Drozd', position: 'CEO & Managing Director', img: CEO },
];

it('renders elem correctly', () => {
    const { getByTestId } = render(<PhotoIcon ourTeam={ourTeam} />);
    expect(getByTestId('photo-icon')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<PhotoIcon ourTeam={ourTeam} />).toJSON();
    expect(tree).toMatchSnapshot();
});
