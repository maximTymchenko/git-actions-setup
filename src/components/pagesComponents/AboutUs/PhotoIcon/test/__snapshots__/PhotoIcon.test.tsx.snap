// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="photo-icon__container"
  data-testid="photo-icon"
>
  <div
    className="photo-icon"
  >
    <div
      className="photo-icon__wrapper"
    >
      <img
        alt="CEO"
        className="photo-icon__img"
        src="test-file-stub"
      />
    </div>
    <span
      className="photo-icon__name"
    >
      Yurii Drozd
    </span>
    <span
      className="photo-icon__position"
    >
      CEO & Managing Director
    </span>
  </div>
  <div
    className="photo-icon"
  >
    <div
      className="photo-icon__wrapper"
    >
      <img
        alt="CEO"
        className="photo-icon__img"
        src="test-file-stub"
      />
    </div>
    <span
      className="photo-icon__name"
    >
      Yurii Drozd
    </span>
    <span
      className="photo-icon__position"
    >
      CEO & Managing Director
    </span>
  </div>
  <div
    className="photo-icon"
  >
    <div
      className="photo-icon__wrapper"
    >
      <img
        alt="CEO"
        className="photo-icon__img"
        src="test-file-stub"
      />
    </div>
    <span
      className="photo-icon__name"
    >
      Yurii Drozd
    </span>
    <span
      className="photo-icon__position"
    >
      CEO & Managing Director
    </span>
  </div>
  <div
    className="photo-icon"
  >
    <div
      className="photo-icon__wrapper"
    >
      <img
        alt="CEO"
        className="photo-icon__img"
        src="test-file-stub"
      />
    </div>
    <span
      className="photo-icon__name"
    >
      Yurii Drozd
    </span>
    <span
      className="photo-icon__position"
    >
      CEO & Managing Director
    </span>
  </div>
  <div
    className="photo-icon"
  >
    <div
      className="photo-icon__wrapper"
    >
      <img
        alt="CEO"
        className="photo-icon__img"
        src="test-file-stub"
      />
    </div>
    <span
      className="photo-icon__name"
    >
      Yurii Drozd
    </span>
    <span
      className="photo-icon__position"
    >
      CEO & Managing Director
    </span>
  </div>
</div>
`;
